### ISSUE TITLE
Lint option: Highlighting copies

### ISSUE THREAD

**Issue by qwerty01** [rust-analyzer/rust-analyzer#7379]
It would be nice to have a lint feature that, when enabled, will highlight all instances where a variable is copied. This would be useful because it's not always clear when a copy would occur, and highlighting copies could point out areas where code compiles correctly, but could be written better.

For example:
```rust
#[derive(Debug, Clone, Copy)]
struct Test {
    num: [u8; 20],
}

fn func(_: Test) {
    // ...
}

fn main() {
    let t = Test{
        num:[0;20],
    };
    let t2 = t.clone();
    func(t);
    println!("{:?}", t2);
}
```

In this case, the author of the code is unaware that `Test` implements `Copy`, so they clone `t` to pass to `func()` in order to keep using `t` after they assume it was moved. When highlighting is enabled, the `t` in `func(t)` would be highlighted which would provide a visual indicator that the clone is unnecessary, as `t` will be copied instead of moved when passed to `func`.

**Comment by Veykril** [comment]
Note that clippy gives you a lint for this as well
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=95f05e8d9bf217e265f51094e192110b
```
warning: using `clone` on type `Test` which implements the `Copy` trait
```

**Comment by qwerty01** [comment]
Depending on how it's implemented, it might also be worth adding a sub-option to disable highlighting for primitive types, since most people would already reason that primitive types implement `Copy` and would find the extra highlighting for them to be unneeded noise.

**Comment by flodiebold** [comment]
I don't think this is currently really actionable, since it requires move analysis to do properly.
